includes/polling/functions.inc.php

```diff
@@ -626,17 +630,26 @@ function json_app_get($device, $extend, $min_version = 1)
 {
     $output = snmp_get($device, 'nsExtendOutputFull.' . string_to_oid($extend), '-Oqv', 'NET-SNMP-EXTEND-MIB');
 
+    // save for returning if not JSON
+    $orig_output = $output;
+
     // make sure we actually get something back
     if (empty($output)) {
         throw new JsonAppPollingFailedException('Empty return from snmp_get.', -2);
     }
 
+    // checks for base64 decoding and converts it to non-base64 so it can gunzip
+    if (preg_match('/^[A-Za-z0-9\/\+\n]+\=*\n*$/', $output)) {
+        $output = base64_decode($output);
```

Comment:
should probably add the strict flag and check the result

Reply:
No need for strict. The regexp takes care of that.

But aye, possible it was improperly generated and is malformed outside of improper characters.